specs/protocol/granite/predeploys.md

```diff
@@ -8,9 +8,115 @@
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
+[stage-1]: https://ethereum-magicians.org/t/proposed-milestones-for-rollups-taking-off-training-wheels/11571#stage-1-limited-training-wheels-3
+
+## Overview
+
+The OP Stack has a permissive management system for L2 predeploys that makes it
+difficult for an OP Stack chain to be considered a [Stage 1 Rollup][stage-1].
+This upgrade is meant to solve the problem of chain operators being able to
+easily break invariants important to the system by being able to arbitrarily
+modify the predeploys themselves.
+
+The `ProxyAdmin` on L2 is upgraded to the `L2ProxyAdmin`, which removes
+the code for handling the L1 legacy proxies and delineates cleanly between
+system ownership and chain operator ownership of different predeploys.
+
+## Constants
+
+| Constant          | Value                           | Description |
+| ----------------- | ------------------------------- | ----------- |
+| `PREDEPLOY_COUNT` |  2048 | The total number of predeploys on L2 |
+| `L2ProxyAdmin` address | `0x4200000000000000000000000000000000000018` | The predeploy of the `ProxyAdmin` on L2 |
+| `DEPOSITOR_ACCOUNT` | `address(0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0001)` | Account that represents the system |
+| `COLLECTIVE_ACCOUNT` | `address(0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0002)` | Account that represents the collective |
+| `CHAIN_ACCOUNT` | `address(0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0003)` | Account that represents the chain operator |
+
+## System Predeploys
+
+System predeploys are predeploys that can only be modified via protocol upgrades. This means
+that the chain governance cannot modify them at the application layer. These predeploys
+are important for maintaining protocol level invariants and therefore the security of
+the system.
+
+TODO: define range of system predeploys
+
+## Userland Predeploys
+
+Userland predeploys are predeploys that can be modified by governance. They should not
+be responsible for managing the security of the system, meaning that if governance was
+to go rogue, then the system should still be considered secure.
+
+Userland predeploys are broken down into two categories, collective governed and chain
+governed.
+
+TODO: define range of userland predeploys, both collective and chain
+
+### Collective Governed
+
+The collective governed userland predeploys are managed at the superchain level. This means
+that the governance of the superchain as a whole is responsible for upgrading these predeploys.
+
+### Chain Governed
+
+The chain governed userland predeploys are managed on a per chain level. This is a commitment
+that these predeploys will never be used by the protocol or superchain governance and can be
+safely used by the chain operator for whatever purposes they desire.
```

Comment:
I like the distinction between chain governed and governance governed—but what predeploys that we currently have would fall into this category? If we don't currently have any, we should just remove this for now and simplify the spec

Reply:
I would prefer to define all existing predeploys as chain governed but would be open for discussion on it. I believe that the `L2ToL1MessagePasser` MUST be chain governed at the least, as well as upcoming interop factory contracts to guarantee solvency of superchain erc20 tokens

If there is a strong argument for particular predeploys being categorized in a specific way, I am open to hearing the argument, but I think given the types described in this document, we have a good framework for categorizing them